signer/src/storage/postgres.rs

```diff
@@ -361,10 +396,74 @@ impl super::DbRead for PgStore {
 
     async fn get_pending_withdraw_requests(
         &self,
-        _chain_tip: &model::BitcoinBlockHash,
-        _context_window: usize,
+        chain_tip: &model::BitcoinBlockHash,
+        context_window: i32,
     ) -> Result<Vec<model::WithdrawRequest>, Self::Error> {
-        Ok(Vec::new()) // TODO(246): Write query + integration test
+        let stacks_chain_tip = self.get_stacks_chain_tip(chain_tip).await?;
+        sqlx::query_as!(
+            model::WithdrawRequest,
+            r#"
+            WITH RECURSIVE extended_context_window AS (
+                SELECT block_hash, parent_hash, confirms, created_at, 1 AS depth
+                FROM sbtc_signer.bitcoin_blocks
+                WHERE block_hash = $1
+                UNION ALL
+                SELECT
+                    parent.block_hash
+                  , parent.parent_hash
+                  , parent.confirms
+                  , parent.created_at
```

Comment:
Looks like these `created_at` columns are unnecessary.

Reply:
Nice observation. I'll clear it out.

Reply:
Done.